### ISSUE TITLE
import TBX? Or any kind of tab-delimited format?

### ISSUE THREAD

**Issue by michaelbeijer** [gaois/terminologue#99]
Hi everyone,

Is it possible yet to import TBX? Or any kind of tab-delimited format? And if so, how?

Michael

**Comment by somethingdark** [comment]
I'm also interested in finding this out. Are there any updates on this matter?

J.

**Comment by michaelbeijer** [comment]
Hmm, I don't think anyone is monitoring this area.

**Comment by somethingdark** [comment]
Hi Michael, 
this is the e-mail I received when I asked for support on this matter. The work around described below did work for me. Until an TBX import is implemented you could try this:

E-Mail extract:
-------------------------------------------------------------
We hope to finish implementing the web-based TBX import function under 'Configuration' very soon, this quarter I hope.

In the meantime, it is possible to convert a TBX file to a Terminologue SQLITE file using the tbx-import.js script. Here are some instructions (Windows):

- Download the Terminologue software.
- Install Node.js and add it to your system path.
- Create a Terminologue formatted Termbase via the Terminologue web interface (Either using your own instance of Terminologue or by creating an account on our cloud-based instance [http://www.terminologue.org<http://www.terminologue.org]www.terminologue.org<http://www.terminologue.org>.)
- Download this termbase (Configuration > Download termbase.sqlite).
- Open 'terminologue\exportimport\tbx-import.js' in an editor, assign the path to your TBX file to the TBXFILE variable, assign the path to your SQLITE file to the SQLITEFILE variable, and save.
- Open a command prompt, navigate to 'terminologue\exportimport', and run 'node tbx-import.js'. This is dependent on 'terminologue\shared\tbx-to-entry.js'.
- Upload your SQLITE file to your Terminologue termbase (Configuration > Upload termbase.sqlite).

-------------------------------------------------------------------------------------------------------

**Comment by somethingdark** [comment]
Nope, sorry can't really help you there, mate. I've been importing terms into clean databases so far. Just give it a try, the process is pretty quick.

**Comment by leplesh** [comment]
Just tested to see what happens with duplicate terms and they are all added. There is no validation. 

For example, if you add an existing term with an identical term ID, it will just add it and incrementally assign it a new ID. So the TBX file must ONLY contain new terms otherwise it becomes a mess pretty quickly. This also suggests that the term IDs in the TBX are irrelevant since they are not validated. Imported terms are simply assigned a new term ID incrementally.

Hope this helps!
